### PR TITLE
refine new api for object detection examples

### PR DESCRIPTION
**Pull request by Spycsh**
## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description 
JIRA ticket: [ticket](https://jira.devtools.intel.com/browse/ILITV-2467)

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: examples/pytorch/object_detection/maskrcnn/quantization/ptq/fx/pytorch/tools/test_net.py [hunk 1]
```diff
@@ -159,14 +159,13 @@ def eval_func(q_model):
...
-        quantizer.model = common.Model(model,
-                                       **{'prepare_custom_config_dict': prepare_custom_config_dict}
-                                      )
-        quantizer.calib_dataloader = cal_dataloader
-        quantizer.eval_func = eval_func
-        q_model = quantizer.fit()
+        from neural_compressor import quantization
+        from neural_compressor.experimental import common
+        from neural_compressor.config import PostTrainingQuantConfig
+        model = common.Model(model, **{'prepare_custom_config_dict': prepare_custom_config_dict})
```
Comment (ftian1) [inline examples/pytorch/object_detection/maskrcnn/quantization/ptq/fx/pytorch/tools/test_net.py:165]: why ** here?

File: examples/pytorch/object_detection/maskrcnn/quantization/ptq/fx/pytorch/tools/test_net.py [hunk 2]
```diff
@@ -159,14 +159,13 @@ def eval_func(q_model):
...
+        conf = PostTrainingQuantConfig(approach="static", backend="pytorch_fx", calibration_sampling_size=[50])
```
Comment (ftian1) [inline examples/pytorch/object_detection/maskrcnn/quantization/ptq/fx/pytorch/tools/test_net.py:166]: backend semantic should have been changed, pls be aware of this and double confirm with penghui

File: examples/pytorch/object_detection/ssd_resnet34/quantization/ptq/fx/python/main.py [hunk 3]
```diff
@@ -586,10 +587,25 @@ def benchmark(model):
...
+        import torchvision.datasets as dset
+        import torchvision.transforms as transforms
+        import torch
+        dataset = dset.CocoDetection(root=args.dataset_path + "/val2017",
+                                     annFile=args.dataset_path +
+                                     "/annotations/instances_val2017.json",
+                                     transform=transforms.ToTensor())
+        dataloader = torch.utils.data.DataLoader(dataset, batch_size=1)
+
+        tuning_criterion = TuningCriterion(timeout=0, max_trials=600)
```
Comment (ftian1) [inline examples/pytorch/object_detection/ssd_resnet34/quantization/ptq/fx/python/main.py:599]: default value of input parameter should be removed for simplicity.

**Comment by chensuyue** [comment]
Pls update benchmark part as well, refer to https://github.com/intel/neural-compressor/pull/174/files

**Comment by chensuyue** [comment]
Please update test json file, so I can test for new API.

File: examples/.config/model_params_pytorch.json [hunk 4]
```diff
@@ -225,15 +225,6 @@
       "batch_size": 100,
       "new_benchmark": false
     },
-    "yolo_v3": {
```
Comment (chensuyue) [inline examples/.config/model_params_pytorch.json:228]: Please do not delete, we will test with old API.

File: examples/.config/model_params_pytorch.json [hunk 5]
```diff
@@ -706,37 +697,22 @@
...
     "ssd_resnet34_qat_fx": {
       "model_src_dir": "object_detection/ssd_resnet34/quantization/qat/fx",
       "dataset_location": "/tf_dataset2/datasets/coco2017/coco",
       "input_model": "/tf_dataset2/models/pytorch/ssd_resnet34/ssd_resnet34_qat_fx.pt",
-      "yaml": "conf.yaml",
-      "strategy": "basic",
-      "batch_size": 1,
-      "new_benchmark": false
-    },
-    "maskrcnn_fx": {
```
Comment (chensuyue) [inline examples/.config/model_params_pytorch.json:732]: Please do not deleted.

File: examples/pytorch/object_detection/ssd_resnet34/quantization/ptq/fx/README.md [hunk 6]
```diff
@@ -81,7 +71,7 @@ GCC5 or above is needed.
```
Comment (chensuyue) [inline examples/pytorch/object_detection/ssd_resnet34/quantization/ptq/fx/README.md:71]: please update in readme: --mode=benchmark --> performance.

File: examples/.config/model_params_pytorch.json [hunk 7]
```diff
@@ -706,37 +697,22 @@
       "model_src_dir": "object_detection/ssd_resnet34/quantization/ptq/fx",
       "dataset_location": "/tf_dataset2/datasets/coco2017/resized_1200x1200/coconpy",
       "input_model": "/tf_dataset2/models/pytorch/ssd_resnet34/resnet34-ssd1200.pytorch",
-      "yaml": "conf.yaml",
-      "strategy": "basic",
-      "batch_size": 1,
-      "new_benchmark": false
+      "main_script": "python/main.py",
+      "batch_size": 1
```
Comment (chensuyue) [inline examples/.config/model_params_pytorch.json:701]: Can we use bigger bs, so run accuracy will be faster.

Comment (Spycsh) [reply examples/.config/model_params_pytorch.json:710]: If we change here to bigger bs, where should we set the batch_size=1 for performance? It is unclear to me

**Comment by chensuyue** [comment]
[extension test](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3806/)

**Comment by chensuyue** [comment]
> [extension test](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3806/)

Please check the tuning regression of ssd_resnet34_fx.

**Comment by chensuyue** [comment]
[retest after bug fix](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3858/)

**Comment by chensuyue** [comment]
[retest after bug fix](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3867/)

**Comment by chensuyue** [comment]
[retest after bug fix](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3876/)

**Comment by chensuyue** [comment]
[retest after bug fix](https://inteltf-jenk.sh.intel.com/job/intel-lpot-validation-top-mr-extension/3880/)

**Review by PenghuiCheng — APPROVED** [review]

**Review by PenghuiCheng — APPROVED** [review]
